src/modules/esl-share/core/esl-share-config.ts

```diff
@@ -32,31 +34,91 @@ export interface ESLShareGroupConfig {
 }
 
 /** The definition of `ESLShare` component configuration */
-export interface ESLShareConfig {
+export interface ESLShareConfigShape {
   /** List of sharing buttons configuration */
   buttons: ESLShareButtonConfig[];
   /** List of share button groups configurations */
   groups: ESLShareGroupConfig[];
 }
 
+/** Gets item of the section by name */
 function getConfigSectionItem<T extends ESLShareButtonConfig | ESLShareGroupConfig>(section: T[], name: string): T | undefined {
   return section.find((item) => item.name === name);
 }
 
-/**
- * Selects the buttons for the given list and returns their configuration.
- * @returns config of buttons
- */
-export function selectButtonsForList(config: ESLShareConfig, list: string): ESLShareButtonConfig[] {
-  return list.split(' ').reduce((res, item) => {
-    const [btnName, groupName] = item.split('group:');
-    if (groupName) {
-      const groupConfig = getConfigSectionItem(config.groups, groupName);
-      if (groupConfig) return res.concat(selectButtonsForList(config, groupConfig.list));
-    } else {
-      const btnConfig = getConfigSectionItem(config.buttons, btnName);
-      if (btnConfig) res.push(btnConfig);
-    }
-    return res;
-  }, [] as ESLShareButtonConfig[]);
+/** Class for managing share buttons configurations */
+export class ESLShareConfig {
+  protected static _config: Promise<ESLShareConfig> = Promise.resolve(ESLShareConfig._from(defaultConfig));
+
+  /**
+   * Creates `ESLShareConfig` instance from the passed object with share component configuration
+   * @returns config instance
+   */
+  protected static _from(cfg: ESLShareConfigShape): ESLShareConfig {
+    return new ESLShareConfig(cfg);
+  }
+
+  /**
+   * Gets promise with buttons config instance.
+   * @returns Promise of the current config instance
+   */
+  public static get(): Promise<ESLShareConfig> {
+    return ESLShareConfig._config;
+  }
+
+  /**
+   * Sets config with a promise of a new config object or using a config provider function.
+   * @returns Promise of the current config
+   */
+  public static set(provider?: ESLShareConfigProviderType | ESLShareConfigShape): Promise<ESLShareConfig> {
+    if (typeof provider === 'function') ESLShareConfig._config = provider().then(ESLShareConfig._from);
+    if (typeof provider === 'object') ESLShareConfig._config = Promise.resolve(ESLShareConfig._from(provider));
+    return ESLShareConfig.get();
+  }
```

Comment:
It doesn't look like we support the primary use case for consumers:
We definitely can provide default settings for buttons, but we 100% can not provide the groups for the end user
So we need a merging strategy instead

Reply:
This has also been changed. You can now add buttons and groups to the config, or override an existing one.
That is, it is quite possible in runtime, for example, to override the group of buttons depending on the locale.